src/map/ai/states/item_state.cpp

```diff
@@ -0,0 +1,303 @@
+}
+
+CItemUsable* CItemState::GetItem()
+{
+    return m_PItem;
+}
+
+void CItemState::InterruptItem(action_t& action)
+{
+    TryInterrupt(dynamic_cast<CBattleEntity*>(GetTarget()));
+
+    if (m_interrupted)
+    {
+        action.id = m_PEntity->id;
+        action.actiontype = ACTION_ITEM_INTERRUPT;
+
+        actionList_t& actionList = action.getNewActionList();
+        actionList.ActionTargetID = (m_PEntity->IsValidTarget(m_targid, m_PItem->getValidTarget(), m_errorMsg) ? GetTarget()->id : 0);
+
+        actionTarget_t& actionTarget = actionList.getNewActionTarget();
+
+        actionTarget.reaction = REACTION_NONE;
+        actionTarget.speceffect = SPECEFFECT_NONE;
+        actionTarget.animation = 54;
+        actionTarget.param = 0;
+        actionTarget.messageID = 0;
+        actionTarget.knockback = 0;
+
+        auto PChar = static_cast<CCharEntity*>(m_PEntity);
+        PChar->pushPacket(m_errorMsg.release());
+
+        PChar->pushPacket(new CInventoryAssignPacket(m_PItem, INV_NORMAL));
+        PChar->pushPacket(new CInventoryItemPacket(m_PItem, m_PItem->getLocationID(), m_PItem->getSlotID()));
+    }
+}
+
+void CItemState::FinishItem(action_t& action)
+{
+    auto PTarget = static_cast<CBattleEntity*>(GetTarget());
+    auto PChar = static_cast<CCharEntity*>(m_PEntity);
+
+    if (m_PItem->getAoE())
+    {
+        PTarget->ForParty([this, PChar, PTarget](CBattleEntity* PMember)
+        {
+            if (!PMember->isDead() && distance(PTarget->loc.p, PMember->loc.p) <= 10)
+            {
+                luautils::OnItemUse(PMember, GetItem());
+            }
+        });
+    }
+    else
+    {
+        luautils::OnItemUse(PTarget, m_PItem);
+    }
+
+    if (m_PItem->isType(ITEM_ARMOR))
+    {
+        if (m_PItem->getMaxCharges() > 1)
+        {
+            m_PItem->setCurrentCharges(m_PItem->getCurrentCharges() - 1);
+        }
+        m_PItem->setLastUseTime(CVanaTime::getInstance()->getVanaTime());
+
+        int8 extra[sizeof(m_PItem->m_extra) * 2 + 1];
+        Sql_EscapeStringLen(SqlHandle, extra, (const char*)m_PItem->m_extra, sizeof(m_PItem->m_extra));
+
+        const int8* Query =
+            "UPDATE char_inventory "
+            "SET extra = '%s' "
+            "WHERE charid = %u AND location = %u AND slot = %u;";
+
+        Sql_Query(
+            SqlHandle,
+            Query,
+            extra,
+            PChar->id,
+            m_PItem->getLocationID(),
+            m_PItem->getSlotID());
+
+        if (m_PItem->getCurrentCharges() != 0)
+        {
+            PChar->PRecastContainer->Add(RECAST_ITEM, m_PItem->getSlotID(), m_PItem->getReuseTime() / 1000);
+        }
+    }
+    else // разблокируем все предметы, кроме экипирвоки
+    {
+        m_PItem->setSubType(ITEM_UNLOCKED);
+
+        charutils::UpdateItem(PChar, m_PItem->getLocationID(), m_PItem->getSlotID(), -1);
+    }
+
+    action.actionid = m_PEntity->id;
```

Comment:
this should be action.id = m_PEntity->id I think, and I think actionid is the item ID